### PR TITLE
PP-1 - Navbar

### PR DESCRIPTION
**Pull request by xfvelocity**
- Added `react-router` to allow for pages/routing
- Added `PPIcon` to allow for more customisation of icons
- Added `Navbar` component with progress bar, back button & custom slot
- Updated so variables can be used across scss files
Screenshot 2023-09-08 212256

### CODE REVIEW

File: src/components/basic/button/PPButton.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import "./ppButton.scss";
 
 // ** Components **
-import PPLoading from "../loading/PPLoading";
+import PPLoading from "@/components/basic/loading/PPLoading";
```
Comment (Docx2) [inline src/components/basic/button/PPButton.tsx:5]: What does @ do

File: src/components/basic/icon/PPIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+const PPIcon = ({ src, className, size = 24 }: Props) => {
+  return (
+    <ReactSVG
+      className={`pp-icon ${className}`}
+      src={`icons/${src}.svg`}
+      style={
+        {
+          "--height": `${size}px`,
```
Comment (Docx2) [inline src/components/basic/icon/PPIcon.tsx:20]: Could have just 1 variable "size"

File: src/components/navbar/navbar.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
+.navbar {
+  border-bottom: 1px solid map-get($pp-colours, "grey-lighten-3");
+  padding: 15px;
+  box-shadow: 0px 3px 4px 0px rgba(0, 0, 0, 0.1);
```
Comment (Docx2) [inline src/components/navbar/navbar.scss:4]: We have to decide on box shadow for nav-bar and footer

File: src/components/navbar/navbar.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
+.navbar {
+  border-bottom: 1px solid map-get($pp-colours, "grey-lighten-3");
```
Comment (Docx2) [inline src/components/navbar/navbar.scss:2]: what does map-get do?

**Review by Docx2 — CHANGES_REQUESTED** [review]

File: src/components/basic/button/PPButton.tsx (hunk 1 above)
Comment (xfvelocity) [reply src/components/basic/button/PPButton.tsx:5]: Points to the `src` folder

File: src/components/navbar/navbar.scss (hunk 3 above)
Comment (xfvelocity) [reply src/components/navbar/navbar.scss:4]: Once they're both merged i'll create a variable & update it for them to match

**Pull request merged**